src/natcap/invest/urban_nature_access.py

```diff
@@ -263,103 +289,161 @@ def execute(args):
-            'working_dir': intermediate_dir,
-        },
-        task_name='2SFCA - Convolve population',
-        target_path_list=[file_registry['convolved_population']],
-        dependent_task_list=[
-            kernel_creation_task,
-            population_alignment_task,
-        ])
-
-    greenspace_nodata = pygeoprocessing.get_raster_info(
-        file_registry['greenspace_area'])['nodata'][0]
-    population_nodata = pygeoprocessing.get_raster_info(
-        file_registry['convolved_population'])['nodata'][0]
-    greenspace_population_ratio_task = graph.add_task(
-        pygeoprocessing.raster_calculator,
-        kwargs={
-            'base_raster_path_band_const_list': [
-                (file_registry['greenspace_area'], 1),
-                (file_registry['convolved_population'], 1),
-                (greenspace_nodata, 'raw'),
-                (population_nodata, 'raw')],
-            'local_op': _greenspace_population_ratio,
-            'target_raster_path': file_registry['greenspace_population_ratio'],
-            'datatype_target': gdal.GDT_Float32,
-            'nodata_target': FLOAT32_NODATA
-        },
-        task_name='2SFCA: Calculate R_j greenspace/population ratio',
-        target_path_list=[file_registry['greenspace_population_ratio']],
-        dependent_task_list=[
-            convolved_population_task,
-            greenspace_reclassification_task,
-        ])
-
-    convolved_greenspace_population_ratio_task = graph.add_task(
-        pygeoprocessing.convolve_2d,
+            task_name=f'Convolve population - {search_radius_m}m',
+            target_path_list=[convolved_population_paths[search_radius_m]],
+            dependent_task_list=[
+                kernel_tasks[search_radius_m],
+                population_alignment_task,
+            ])
+
+    greenspace_attrs = attr_table[attr_table['greenspace'] == 1]
+    search_radii = greenspace_attrs['search_radius_m'].unique()
+    if len(search_radii) == 1:
+        LOGGER.info(
+            'Only 1 greenspace search radius found. '
+            'Running 2SFCA in unified greenspace mode, calculating '
+            'greenspace supply once for all LULC classes.')
+
+        # If all of the search radii are the same, then we run 2SFCA with all
+        # of the greenspace lucodes in one batch.
+        radius_to_lucodes = [
+            (search_radii[0], greenspace_attrs['lucode'].unique())]
+    else:
+        LOGGER.info(
+            'Multiple search radii found. '
+            'Running 2SFCA in split greenspace mode, calculating greenspace '
+            'supply per LULC class.')
+
+        # If there are varying search radii, we run 2SFCA once for each
+        # radius:lucode combination so that modelers can see the contribution
+        # of each landcover code to the total greenspace supply.  This comes at
+        # the cost of more convolutions and more runtime (and disk space), but
+        # it's what the UNA science team wants.
```

Comment:
```suggestion
        # the cost of more convolutions and more runtime (and disk space).
```

Reply:
I suggested a removal of that last bit because I'm not sure it's helpful. Let me know what you think!